src/controller.py

```diff
@@ -70,7 +72,8 @@ def test_lab(self, current_user, lab_id, lab_src_url, revision_tag=None):
                     self.update_deploy_record(current_user)
                     self.state.save(self.deploy_record.record)
                     logger.info("test_lab(): test succcessful, ip = %s" % ip)
-                    return ip
+                    domain_name = self.register_lab(lab_id, ip)
+                    return domain_name
```

Comment:
Did you think how this will work if ads is being used in developer environment?

Provide a way in confing.py to set a flag to use hooks or not and use that flag before register is invoked.